### ISSUE TITLE
Steep slopes on junior coaster are only available when 'select rides by track type' is deselected

### ISSUE THREAD

**Issue by RoelerCoaster** [OpenRCT2/OpenRCT2#5902]
**OS:** Windows 10
**Version:** 0.1.1-develop
**Commit/Build:** ff51d4a

The steep slopes for the junior coaster only seem to be available when the option 'select rides by track type' is _not_ selected. From #5691 I understand that the steep tracks should only be available for certain (custom) train types and not for the default RCT2 ones. From #5182 I understand that the 'select rides by track type' can override this. 

The behaviour, however, seems to be the wrong way around, as there are steep slopes available when using the RCT2-junior coaster with 'select rides by track type' disabled (and not available when enabled)

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Multiplayer?

**Steps to reproduce:**
1. Open a scenario with the junior coaster available
2. Deselect 'select rides by track type' in the options
3. Steep slopes are available when building the rollercoaster
4. Select 'select rides by track type'
5. The steep slopes are no longer available.

**Comment by rwjuk** [comment]
Tagging @Gymnasiast as he worked on the implementation of this.

**Comment by Gymnasiast** [comment]
The problem is that the .DAT files of the Ladybird and Log trains that come with RCT2 tell the game that they can handle steep inclines. When using the regular mode, the game will use this information to determine which pieces to enable (a piece will get enabled if both the track itself and the train say it is supported). This was never a problem in vanilla RCT2 as it didn't support steep inclines on the Junior RC at all.

When you put the game in select-by-track-type mode, the game will ignore what the train says it's capable of. Instead it enables the track pieces of the so-called _ride group_ the train belongs to. There are two for the Junior RC: one is called 'Junior Roller coaster', the other is called 'Classic Mini Roller Coaster'. All trains that come with RCT2 and its expansions are examples of the former, and thus will not show steep inclines. Some custom-made trains do support steep inclines, and will thus show them.

So, in short, it's a data file error that has been exposed by OpenRCT2. We already apply fixes to many vanilla objects and I will therefore add fixes for the Ladybird, Log, Witchity Grub and Panda trains.
